src/com/goide/formatter/GoOnSaveFormatter.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Mihai Toader, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.formatter;
+
+import com.goide.GoFileType;
+import com.goide.actions.fmt.GoFmtFileAction;
+import com.goide.actions.fmt.GoImportsFileAction;
+import com.goide.settings.GoApplicationSettings;
+import com.intellij.execution.ExecutionException;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileDocumentManagerAdapter;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import org.jetbrains.annotations.NotNull;
+
+
+public class GoOnSaveFormatter extends FileDocumentManagerAdapter {
+  @Override
+  public void beforeDocumentSaving(@NotNull final Document document) {
+
+    if (!document.isWritable()) {
+      return;
+    }
+
+    final Project[] projects = ProjectManager.getInstance().getOpenProjects();
+    if (projects.length == 0) {
+      return;
+    }
+
+    final VirtualFile file = FileDocumentManager.getInstance().getFile(document);
+    if (file == null || file.getFileType() != GoFileType.INSTANCE) {
+      return;
+    }
+
+    Project project = null;
+    for (Project possibleProject : projects) {
```

Comment:
Don't use source/content roots ever, in go we can hardly manage it and using it leads to problems with different go-project workflow